src/main/java/org/gradle/java/JigsawPlugin.java

```diff
@@ -66,15 +70,10 @@ private void configureJavaTasks(Project project) {
         project.afterEvaluate(new Action<Project>() {
             @Override
             public void execute(Project project) {
-                if (project.getPlugins().hasPlugin(JAVA_PLUGIN)
-                        || project.getPlugins().hasPlugin(JAVA_LIBRARY_PLUGIN)) {
-                    configureCompileJavaTask(project);
-                    configureCompileTestJavaTask(project);
-                    configureJarTask(project);
-                    configureTestTask(project);
-                } else {
-                    throw new GradleException("You must apply the java or java-library plugin as well.");
-                }
+                configureCompileJavaTask(project);
+                configureCompileTestJavaTask(project);
+                configureJarTask(project);
+                configureTestTask(project);
                 if (project.getPlugins().hasPlugin(APPLICATION_PLUGIN)) {
```

Comment:
:x: this should be replaced with:

```
project.getPluginManager().withPlugin("application", {
    // do something to configure the application plugin tasks
});
```
which would be applied if and only if the application plugin is applied, independently of when.